packages/cardano-services/src/ChainHistory/ChainHistoryHttpService.ts

```diff
@@ -0,0 +1,91 @@
+import * as OpenApiValidator from 'express-openapi-validator';
+import { Awaited } from '../types';
+import { ChainHistoryProvider, ProviderError, ProviderFailure } from '@cardano-sdk/core';
+import { DbSyncChainHistoryProvider } from './DbSyncChainHistoryProvider/DbSyncChainHistory';
+import { HttpServer, HttpService } from '../Http';
+import { Logger, dummyLogger } from 'ts-log';
+import { ServiceNames } from '../Program';
+import { providerHandler } from '../util';
+import express from 'express';
+import path from 'path';
+
+export interface ChainHistoryHttpServiceDependencies {
+  logger?: Logger;
+  chainHistoryProvider: DbSyncChainHistoryProvider;
+}
+
+export class ChainHistoryHttpService extends HttpService {
+  #chainHistoryProvider: ChainHistoryProvider;
+  private constructor(
+    { logger = dummyLogger, chainHistoryProvider }: ChainHistoryHttpServiceDependencies,
+    router: express.Router
+  ) {
+    super(ServiceNames.ChainHistory, router, logger);
+    this.#chainHistoryProvider = chainHistoryProvider;
+  }
+
+  async healthCheck() {
+    return this.#chainHistoryProvider.healthCheck();
+  }
+
+  static async create({ logger = dummyLogger, chainHistoryProvider }: ChainHistoryHttpServiceDependencies) {
+    const router = express.Router();
+    if (!(await chainHistoryProvider.healthCheck()).ok) {
+      throw new ProviderError(ProviderFailure.Unhealthy);
+    }
+    const apiSpec = path.join(__dirname, 'openApi.json');
+    router.use(
+      OpenApiValidator.middleware({
+        apiSpec,
+        ignoreUndocumented: true, // otherwhise /metrics endpoint should be included in spec
+        validateRequests: true,
+        validateResponses: true
+      })
+    );
+    // TODO: refactor providerHandler to get Args and Response types on its own?
+    //       and use it as providerHandler<typeof chainHistoryProvider.blocksByHashes>
```

Comment:
Is there any reason why we should pass args and response types manually? Does it make sense to change `providerHandler` to be able to use it like this instead?

```ts
providerHandler<typeof chainHistoryProvider.blocksByHashes>(async (args, _, res, handler) => {
``` 

Or even something like this: 
```ts
providerHandler(chainHistoryProvider.blocksByHashes)(async (args, _, res, handler) => {
        try {
          return HttpServer.sendJSON(res, await handler(...args));
        } catch (error) {
          logger.error(error);
          return HttpServer.sendJSON(res, new ProviderError(ProviderFailure.Unhealthy, error), 500);
        }
      }
```

Reply:
No reason, I agree, inferring the types would be better